### ISSUE TITLE
Hotkey Delay Not Applying To Reset Hotkey

### ISSUE THREAD

**Issue by Megamatt79** [LiveSplit/LiveSplit#2200]
Greetings,

I am writing to report an issue that I am experiencing with LiveSplit.  Specifically, my capture card (Elgato Game Capture HD) introduces minor lag when recording/streaming via OBS Studio.  To compensate, I have enabled the Hotkey Delay feature within LiveSplit at 0.72 seconds and it works perfectly with my designated Start/Split hotkey (the spacebar).  However, my Reset hotkey (the Enter key) appears to be working without the desired delay included.

Is this a known issue?  Is there another setting that I may be overlooking that will apply the hotkey delay to the Reset hotkey?  I have attached a snip of my current settings for your consideration.

Any insight that you can provide would be greatly appreciated.  

Thanks in advance,

Matt

![LiveSplit_Settings_MM79_20211214](https://user-images.githubusercontent.com/96421295/146759944-8ec4d7f6-14a0-48d1-a694-fef7e37d9170.PNG)

**Comment by Megamatt79** [comment]
> Hey Megamatt I don't think the hotkey delay effects the reset action as there's generally no reason why one would want to reset their timer with a delay (other than your genuine usecase of course). I think the main idea of that feature was to allow people to compensate for console/capture delays to keep their times properly tracked. When resetting the run, the timer gets stopped and the current split will not be included in any time calculations like best of sum and such. What I would like to know if there's anything that it currently messes up (other than being async to your camera/keyboard sounds)? But you're right, one would assume that it would apply to _all_ hotkeys, not just the one's being relevant for split times.
> 
> Best regards

Thank you very much for the prompt reply!  To answer your question, no, I have not noticed any other issues with the Reset hotkey other than the lack of the specified delay.  

If the hotkey delay feature was intended to function as you noted above (where the Reset hotkey was deliberately made not applicable) and no one else has raised this same concern, then I would be more than happy to let this matter rest.  My bigger concern was that the absence of delay was due to something that I messed up or overlooked in my settings.

**Comment by NoTeefy** [comment]
> Thank you very much for the prompt reply! To answer your question, no, I have not noticed any other issues with the Reset hotkey other than the lack of the specified delay.
> 
> If the hotkey delay feature was intended to function as you noted above (where the Reset hotkey was deliberately made not applicable) and no one else has raised this same concern, then I would be more than happy to let this matter rest. My bigger concern was that the absence of delay was due to something that I messed up or overlooked in my settings.

Okay, so I just took a quick look at the implementation (https://github.com/LiveSplit/LiveSplit/blob/de77a492d96a45d20d748d3d2e832a506747542c/LiveSplit/LiveSplit.View/View/TimerForm.cs#L1076) and my assumption was correct that it only takes the delay into account on the events *split* and *pause*. Everything else ignores the delay.
When one starts their run, it triggers a split event which is why it also delays that specific input.
In conclusion this would mean that you could either suggest the maintainers to extend that check to all input events or make a pull request for it yourself.

Best regards
